### PR TITLE
feat: add meta data to pipeline.

### LINKED ISSUE #39: feat: transport metadata and emit diagnostics

### PR DESCRIPTION
**Pull request by pekasen**
Adds a Record-class with which data is wrapped and meta data are gathered through the pipeline. Adds additional tests for Record.py, NDJSONReader.py and TwacapicReader.py.

Coverage is now 60%

### CODE REVIEW

File: dabapush/Record.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    """
+
+    payload: Dict[str, Any]
+    source: Optional[Any] = None
+    uuid: Optional[str] = uuid4().hex
+    processed_at: datetime = datetime.now()
+    children: List["Record"] = dataclasses.field(default_factory=list)
+
+    def split(self, key: str, id_key: Optional[str] = None) -> List["Record"]:
+        """Split the record based on a key in the payload.
```
Comment (FlxVctr) [inline dabapush/Record.py]: an example would help

**Review by FlxVctr — APPROVED** [review]
looks alright, but untested by me. Would be good to find a way to get the CI checks running on GH

**Comment by pekasen** [comment]
> looks alright, but untested by me. Would be good to find a way to get the CI checks running on GH

Yep, CI testing hangs itself on the dependency to `smo-database`, which is a private git-submodule as of yet. https://github.com/Leibniz-HBI/dabapush/pull/45 removes this and testing is then available in CI.

File: dabapush/Record.py (hunk 1 above)
Comment (pekasen) [reply dabapush/Record.py]: Done in https://github.com/Leibniz-HBI/dabapush/pull/42/commits/59a8879f0f4ef9227df61c5c3b52c8c4cfd19033.

**Pull request merged**